sound/hybrid_font.h

```diff
@@ -230,18 +230,72 @@ class HybridFont : public SaberBase, public Looper {
     if (loop) hum_player_->PlayLoop(loop);
   }
 
+  int saved_player_to_restart_ = -1 ;
+  RefPtr<BufferedWavPlayer> GetOrFreeWavPlayer(Effect* e) {
+    RefPtr<BufferedWavPlayer> free_player = GetFreeWavPlayer();
+    if (!free_player) {
+      STDOUT.println("Out of WAV players! Getting more...");
+      float saved_progress = 0.00;
+      float highest_progress = 0.00;
+      int player_to_restart = -1;
+      int count = 0;
+      // check wavplayers, rule out hum and smoothswings then kill oldest one 
+      for (size_t i = 0; i < NELEM(wav_players); i++) {
+        if (wav_players[i].isPlaying() &&
+          wav_players[i].current_file_id().GetEffect() == e &&
+          wav_players[i].refs() == 0) {
+          count++;
+          float pos = wav_players[i].pos();
+          if (pos > highest_progress) {
+            saved_progress = highest_progress;
+            highest_progress = pos;
+            saved_player_to_restart_ = player_to_restart;
+            player_to_restart = i;
+          } else if (pos > saved_progress) {
+            saved_progress = pos;
+            saved_player_to_restart_ = i;
+          }
+        }
+      }
+      if (count > 1) {
+        STDOUT << "*************** Ok count " << count << "  > 1\n";
+        wav_players[player_to_restart].set_fade_time(0.001);
+        wav_players[player_to_restart].FadeAndStop();
+        STDOUT << "Stopping wav_player " << player_to_restart << " playing " << wav_players[player_to_restart].filename() << "\n";
+        while (wav_players[player_to_restart].isPlaying()) {
+#ifdef VERSION_MAJOR >= 4
+          armv7m_core_yield();
+#endif
+        }
+        free_player = RefPtr<BufferedWavPlayer>(wav_players + player_to_restart);
+      } else {
+        if (saved_player_to_restart_ > -1) {
+          wav_players[saved_player_to_restart_].set_fade_time(0.001);
+          wav_players[saved_player_to_restart_].FadeAndStop();
+          STDOUT << "Stopping wav_player " << saved_player_to_restart_ << " playing " << wav_players[saved_player_to_restart_].filename() << "\n";
+          while (wav_players[saved_player_to_restart_].isPlaying()) {
+#ifdef VERSION_MAJOR >= 4
+            armv7m_core_yield();
+#endif
+          }
+          free_player = RefPtr<BufferedWavPlayer>(wav_players + saved_player_to_restart_);
+        }
+      }
+      return RefPtr<BufferedWavPlayer>(); // failure to fulfill is same lag as just waiting like we did before all this. Not great.
```

Comment:
The comment will go away. Just a personal note really. "same lag as we had before", meaning a dead button while we wait for a free player, which is what all of this was supposed to prevent.

Reply:
There is no way to prevent it in every single case, but the problem here is that the comments makes no sense.
Lag means that a sound has been delayed somehow, not missed, which is what happens in this case.

Reply:
removed